pkg/account/handler.go

```diff
@@ -137,48 +224,97 @@ func (h *AccountHandler) ResetPasswd(c echo.Context) error {
 	}
-	return modifyPasswd(c, Email, Passwd)
+
+	account.Passwd = body.NewPasswd
+	account.HashPassword()
+
+	return c.JSON(http.StatusOK, api.Return("Successfully modified", nil))
 }
 
 /**
  * @brief private method for checking password
  */
-func checkPasswd(c echo.Context, Email string, Passwd string) error {
-	// Travese to find matched account, use DB later
-	for itor := accountList.Front(); itor != nil; itor = itor.Next() {
-		if itor.Value.(Account).Email == Email {
-			if itor.Value.(Account).Passwd == Passwd {
-				return c.JSON(http.StatusOK, api.Return("Successfully logged in", nil))
-			}
-			return c.JSON(http.StatusBadRequest, api.Return("Wrong Password", nil))
-		}
+func checkPasswd(c echo.Context, Email string, Passwd string, account *Account, db *gorm.DB) error {
+	if err := db.Where("email = ?", Email).First(&account).Error; err != nil { // not found
+		return c.JSON(http.StatusBadRequest, api.Return("E-Mail not found", nil))
+	}
+	if bcrypt.CompareHashAndPassword([]byte(account.Passwd), []byte(Passwd)) != nil {
+		return c.JSON(http.StatusBadRequest, api.Return("Wrong Password", nil))
 	}
+	return c.NoContent(http.StatusOK)
+}
 
-	return c.JSON(http.StatusBadRequest, api.Return("E-Mail not found", nil))
+/**
+ * @brief private method for hashing password
+ */
+func (u *Account) HashPassword() {
+	bytes, _ := bcrypt.GenerateFromPassword([]byte(u.Passwd), bcrypt.DefaultCost)
+	u.Passwd = string(bytes)
 }
 
 /**
- * @brief private method for modifying password
+ * @brief private method for generateing cookie token
  */
-func modifyPasswd(c echo.Context, Email string, Passwd string) error {
-	for itor := accountList.Front(); itor != nil; itor = itor.Next() {
-		if itor.Value.(Account).Email == Email {
-			// Remove this and push a new one with new passwd
-			accountList.PushBack(Account{Email: Email, Type: itor.Value.(Account).Type, Name: itor.Value.(Account).Name, Passwd: Passwd})
-			accountList.Remove(itor)
-
-			return c.JSON(http.StatusOK, api.Return("Successfully modified", nil))
+func (u *Account) GenerateToken() (string, error) {
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"id": u.ID,
+	})
+
+	tokenString, err := token.SignedString(jwtKey)
+	return tokenString, err
+}
+
+// Authoriszed : Check Auth
+func Authoriszed(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		cookie, err := c.Cookie("token")
+		if err != nil {
+			return c.NoContent(http.StatusUnauthorized)
 		}
+
+		token, err := jwt.Parse(cookie.Value, func(token *jwt.Token) (interface{}, error) {
+			if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, fmt.Errorf("Unexpected Signing Method: %v", token.Header["alg"])
+			}
+
+			return jwtKey, nil
+		})
+
+		if !token.Valid || err != nil {
+			return c.NoContent(http.StatusUnauthorized)
+		}
+
+		c.Set("username", token.Claims.(jwt.MapClaims)["username"])
+
+		return next(c)
```

Comment:
So we can get `username` from `c echo.context` right?

Reply:
something like this, am working on it